scripts/test_publish.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -x -e
+
+# Usage:
+#   cargo install cargo-local-registry
+#   cargo install cargo-index
+#   grep -v '^#' packages.txt | scripts/test_publish.sh REGISTRY
```

Comment:
* The documentation should indicate what is being tested. I think it is the way that the packages are ordered. But I am not entirely sure, so this deserves mentioning.
* I find the check for the nothing uncommited in git to be just annoying. If you edit your system, you want to do that check at the end.
* Also, important is whether the testing is needed before doing the `publish.sh`. A priori no, but better said than nothing.
* The command modifies the `.cargo/config.toml`. Why so? The construction of the REGISTRY, the git action and so on make it looks than much more than just testing is done.

Reply:
> I find the check for the nothing uncommited in git to be just annoying.

I narrowed down the check to a single file.

> The command modifies the .cargo/config.toml.

It's temporary